src/main/java/com/laytonsmith/core/Script.java

```diff
@@ -884,7 +889,8 @@ public void checkAmbiguous(List<Script> scripts) throws ConfigCompileException {
 						// the same argument position.
 						if(c1.getCType() == c2.getCType()
 								&& (c1.getCType() == ConstructType.STRING || c1.getCType() == ConstructType.COMMAND)) {
-							if(c1.nval() != c2.nval() && (c1.nval() == null || !c1.nval().equals(c2.nval()))) {
+							if(Construct.nval(c1) != Construct.nval(c2) && (Construct.nval(c1) == null
+									|| !Construct.nval(c1).equals(Construct.nval(c2)))) {
```

Comment:
I just realized this is a bug, since nval can return java null, and we're dereferencing it here.

Reply:
Should these Constructs ever be CNull anyway?